Lab_13-15/raport.md

```diff
@@ -553,54 +649,235 @@ FROM us_cities c1, us_cities c2
 WHERE c1.city = 'Buffalo' and c2.city = 'Syracuse';
 ```
 
+> Wyniki, zrzut ekranu, komentarz
 
+![](img/ex7/12.png)
 
->Wyniki, zrzut ekranu, komentarz
+> Odległość między miastami Buffalo i Syracuse wynosi 222.18 km.
+
+Dodatkowo:
+
+a)     Oblicz odległość między miastem Tampa a drogą I4
 
 ```sql
---  ...
+SELECT SDO_GEOM.SDO_DISTANCE ( c1.location, i.geom, 0.5, 'unit=kilometer') distance
+FROM us_cities c1, us_interstates i
+WHERE c1.city = 'Tampa' and i.interstate = 'I4';
 ```
 
-Dodatkowo:
+![](img/ex7/8.png)
 
-a)     Oblicz odległość między miastem Tampa a drogą I4
+> Odleglosc miedzy miastem Tampa a drogą I4 to 3.10 km.
+
+b)    Jaka jest odległość między stanem Nowy Jork a  Florydą
+
+```sql
+SELECT SDO_GEOM.SDO_DISTANCE (s1.geom, s2.geom, 0.5, 'unit=kilometer') distance
+FROM us_states s1, us_states s2
+WHERE s1.state = 'New York' and s2.state = 'Florida';
+```
+
+![](img/ex7/9.png)
+
+> Odległość między stanem Nowy Jork a Florydą wynosi 1256.58 km.
 
-b)    Jaka jest odległość z między stanem Nowy Jork a  Florydą
+c)     Jaka jest odległość między miastem Nowy Jork a Florydą
 
-c)     Jaka jest odległość z między miastem Nowy Jork a  Florydą
+```sql
+SELECT SDO_GEOM.SDO_DISTANCE (c1.location, s2.geom, 0.5, 'unit=kilometer') distance
+FROM us_cities c1, us_states s2
+WHERE c1.city = 'New York' and s2.state = 'Florida';
+```
+
+![](img/ex7/10.png)
+
+> Odległość między miastem Nowy Jork a Florydą to 1296.59 km.
 
 d)    Podaj 3 parki narodowe do których jest najbliżej z Nowego Jorku, oblicz odległości do tych parków
 
+```sql
+SELECT p.name,
+    SDO_GEOM.SDO_DISTANCE(c.location, p.geom, 0.5, 'unit=mile') AS distance
+FROM us_parks p,
+     us_cities c
+WHERE c.city = 'New York'
+AND SDO_NN(p.geom, c.location, 'sdo_num_res=3') = 'TRUE'
+ORDER BY distance
+```
+
+![](img/ex7/11.png)
+
+> Trzy najbliższe parki to Institute Park, Prospect Park oraz Thompkins Park.
+
 e)    Przetestuj działanie funkcji
 
-a.     sdo_intersection, sdo_union, sdo_difference
+    a.     sdo_intersection, sdo_union, sdo_difference
 
-b.     sdo_buffer
+**sdo_intersection**
 
-c.     sdo_centroid, sdo_mbr, sdo_convexhull, sdo_simplify
+```sql
+SELECT SDO_GEOM.SDO_INTERSECTION(a.geom, b.geom, 0.005) AS intersection_geom
+FROM geometry_table a, geometry_table b
+WHERE a.id = 1 AND b.id = 2;
+```
```

Comment:
Maybe there should be some plotted examples of the tests? We have to justify the conclusions and show that we ran the commands

Reply:
I added some pictures from the internet